src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Unix.cs

```diff
@@ -63,11 +61,16 @@ private void AcceptCompletionCallback(IntPtr acceptedFileDescriptor, byte[] sock
 
         internal unsafe SocketError DoOperationAccept(Socket socket, SafeCloseSocket handle, SafeCloseSocket acceptHandle, out int bytesTransferred)
         {
+            if (_buffer != null)
+            {
+                throw new PlatformNotSupportedException(SR.net_sockets_accept_receive_notsupported);
+            }
+
```

Comment:
On Windows, it's just an optimization that limits the number of user-mode/kernel transitions: https://msdn.microsoft.com/en-us/library/windows/desktop/ms737524(v=vs.85).aspx

The AcceptEx function combines several socket functions into a single API/kernel transition. The AcceptEx function, when successful, performs three tasks:
* A new connection is accepted.
* Both the local and remote addresses for the connection are returned.
* The first block of data sent by the remote is received.